msgio.go

```diff
@@ -158,18 +162,27 @@ func NewReaderWithPool(r io.Reader, p *mpool.Pool) ReadCloser {
 // NextMsgLen reads the length of the next msg into s.lbuf, and returns it.
 // WARNING: like Read, NextMsgLen is destructive. It reads from the internal
 // reader.
-func (s *reader) NextMsgLen() (int, error) {
+func (s *reader) NextMsgLen() (uint32, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 	return s.nextMsgLen()
 }
 
-func (s *reader) nextMsgLen() (int, error) {
-	if s.next == -1 {
+func (s *reader) nextMsgLen() (uint32, error) {
+	if s.more {
 		if _, err := io.ReadFull(s.R, s.lbuf); err != nil {
 			return 0, err
 		}
-		s.next = int(NBO.Uint32(s.lbuf))
+		n := NBO.Uint32(s.lbuf)
+		if n < 0 {
```

Comment:
going to remove this? 

was waiting for this to merge, but need my changes (#9) sooner. also cleans up all this logic and puts it in one place-- may be easier to rebase on that than vv